cpg_workflows/metamist.py

```diff
@@ -314,13 +364,15 @@ def update_analysis(self, analysis: Analysis, status: AnalysisStatus):
         """
         Update "status" of an Analysis entry.
         """
-        try:
-            self.aapi.update_analysis(
-                analysis.id,
-                models.AnalysisUpdateModel(status=models.AnalysisStatus(status.value)),
-            )
-        except ApiException:
-            traceback.print_exc()
+        self.make_aapi_call(
+            self.aapi.update_analysis,
+            analysis_id=analysis.id,
+            analysis_update_model=models.AnalysisUpdateModel(
+                status=models.AnalysisStatus(status.value),
+            ),
+        )
+        # Keeping this as is for compatibility with the existing code
+        # However this should only be set after the API call is successful
         analysis.status = status
 
     # NOTE: This isn't used anywhere.
```

Comment:
🙃

Reply:
I suggested we delete the code in a separate PR to not increase the scope here :)